io/exportModel.m

```diff
@@ -222,7 +222,7 @@ function exportModel(model,fileName,exportGeneComplexes,supressWarnings)
     end
     
     if isfield(modelSBML.compartment,'metaid')
-        if ~isnan(str2double(model.comps(i)))
+        if any(regexp(model.comps(i),'^\d'))
```

Comment:
Would it make sense to also avoid other non-letter characters, or is that taken care of at some other point?

```suggestion
        if any(regexp(model.comps(i),'^[^a-zA-Z]'))
```

Reply:
You're right that `^\d` is actually too narrow, but besides letters also underscore is [allowed](https://github.com/SysBioChalmers/RAVEN/issues/353#issuecomment-875004745) at beginning of `SId` in SBML. Would that make `^[a-zA-Z_]` as regex?

Reply:
The first `^` is an anchor for the beginning of string, and the second `^` is for the negative pattern. It is of course possible to remove the negative pattern as suggested above, but then `any` should turn into `~any`. With the suggestion above, a compartment like `no_prefix_be4_no` would be end up prefixed.

Below is the updated suggestion including the accepted `_`, and removing the `any` which is redundant.
```suggestion
        if regexp(model.comps(i),'^[^a-zA-Z_]')
```

Reply:
You're of course correct :), confusing that `^` can have multiple meanings (anchor and negation)!

Reply:
Hehe I had to look it up myself to be sure. I saw somebody use https://regex101.com and now that's my go to when in doubt of regex patterns.
Thanks for looking up the SBML pattern, it would have been so easy to miss that `_`.